### PR TITLE
Call end write stream in updateStoreData and deleteStoreData

### PR DESCRIPTION
**Pull request by advplyr**
Tldr; After the first update to the db, all other updates are failing. The `.old` file was not getting deleted due to the write stream staying open.

````javascript
// utils.js replaceFile
await promisify(rename)(a, a + ".old");  // <---- EPERM Error here on 2nd update.
await promisify(rename)(b, a);
await promisify(unlink)(a + ".old");    // <---- Successful on 1st update, but file is not actually removed
 ````

On Windows 10 using Node v12.19.0 an issue was occurring where the error
`Error: EPERM: operation not permitted, rename`
was being thrown on the second update.

The first update was successful, but it was not unlinking the last `.old` file. The `.old` file remained in the directory but could not be read or deleted until the nodejs process was closed. What was happening was the `.old` file still had an open file handler so when `unlink` was called Windows was "marking the file for deletion" and couldn't be accessed until the file handler was closed. [Explained here](https://stackoverflow.com/a/27277865/7431543)

After a lot of trial and error I found that the write stream in `updateStoreData` was never closing. This appears to be Windows specific, because I run njodb in linux with no issues. I think the best way to ensure the write stream gets closed is by calling the `.end()` method on the write stream when the read stream gets closed. Once the write stream is closed the `.old` file gets unlinked properly.

For testing I added `.on("close", ...)` listeners to the write stream with a few console logs and found that when calling `.end()` the write stream closed properly.

Also updated a small typo in `objects.js`

### CODE REVIEW

**Comment by jamesbontempo** [comment]
Hey there! I apologize for not getting around to this sooner. I've been a bit swamped so haven't even had a chance to look at this yet :( But I will hopefully get some time to do so in the next few days...

**Comment by advplyr** [comment]
No problem, it's not urgent at all. This is only effecting my dev environment, so I'm using my fork for now.

**Comment by advplyr** [comment]
I want to add that this issue also occurs inside a docker container running on Windows 10.

**Comment by jamesbontempo** [comment]
Thanks for catching that! I've merged your commit 👍

I'm thinking about maybe moving the call to `writer.end()` into the handler's `return` method (the `writer`, if there is one, would be passed in as a parameter), but I'm not entirely sure that's a worthwhile refactoring...
